test/jdk/java/nio/file/spi/SetDefaultProvider.java

```diff
@@ -73,6 +76,45 @@ public void testClassPath() throws Exception {
         assertTrue(exitValue == 0);
     }
 
+    /**
+     * Test override of default FileSystemProvider with a
+     * FileSystemProvider jar and the main application on the class path.
+     */
+    public void testClassPathWithFileSystemProviderJar() throws Exception {
+        String testClasses = System.getProperty("test.classes");
+        Path jar = Path.of("testFileSystemProvider.jar");
+        Files.deleteIfExists(jar);
+        createFileSystemProviderJar(jar, Path.of(testClasses));
+        String classpath = jar + File.pathSeparator + testClasses
+                + File.separator + "modules" + File.separator + "m";
+        int exitValue = exec(SET_DEFAULT_FSP, "-cp", classpath, "p.Main");
+        assertTrue(exitValue == 0);
+    }
+
+    /**
+     * Creates a JAR containing the FileSystemProvider used to override the
+     * default FileSystemProvider
+     */
+    private void createFileSystemProviderJar(Path jar, Path dir) throws IOException {
+
```

Comment:
In this test, the supporting methods are at the end of the source file, probably should keep it consistent.